src/phylo/node_basics.jl

```diff
@@ -0,0 +1,335 @@
+# number of taxa + 1.
+#
+# This file is a part of BioJulia. License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+
+
+
+"""
+    isroot(tree::Phylogeny, vertex::Int)
+
+Test whether a vertex is the root.
+
+# Examples:
+```julia
+isroot(my_tree, 9)
+```
+"""
+function isroot(tree::Phylogeny, vertex::Int)
+    return vertex == root(tree)
+end
+
+
+"""
+    isroot{T <: SymString}(tree::Phylogeny, vertex::T)
+
+Test whether a vertex is the root.
+
+# Examples:
+```julia
+isroot(my_tree, :Root)
+isroot(my_tree, :A)
+```
+"""
+function isroot{T <: SymString}(tree::Phylogeny, vertex::T)
+    isroot(tree, tree.vertexindex[name])
+end
+
+
+
+"""
+    isclade(tree::Phylogeny, vertex::Int)
+
+Test whether a node is internal clade, i.e. has a parent and one or more children.
+
+# Examples:
+```julia
+isclade(my_tree, 6)
+```
+"""
+function isclade(tree::Phylogeny, vertex::Int)
+    return vertex >= root(tree)
+end
+
+
+"""
+    isclade(tree::Phylogeny, vertex::SymString)
+
+Test whether a node is internal, i.e. has a parent and one or more children.
+
+# Examples:
+```julia
+isclade(my_tree, :Node7)
+isclade(my_tree, "Homo Sapiens")
+isclade(my_tree, Symbol("Homo Sapiens"))
+```
+"""
+function isclade(tree::Phylogeny, name::SymString)
+    return isclade(tree, tree.vertexindex[name])
+end
+
+
+
+"""
+    isleaf(tree::Phylogeny, vertex::Int)
+
+Test whether a vertex is a leaf.
+
+A vertex is a leaf, when it has a parent node, but no children nodes.
+
+# Examples:
+```julia
+isleaf(my_phylogeny, 1)
+```
+"""
+function isleaf(tree::Phylogeny, vertex::Int)
+    return vertex <= tree.ntaxa
+end
+
+
+"""
+    isleaf(tree::Phylogeny, name::SymString)
+
+Test whether a vertex is a leaf.
+
+A vertex is a leaf, when it has a parent node, but no children nodes.
+
+# Examples:
+```julia
+isleaf(my_phylogeny, :A)
+isleaf(my_phylogeny, "A")
+```
+"""
+function isleaf(tree::Phylogeny, name::SymString)
+    return isleaf(tree, tree.vertexindex[name])
+end
+
+
+
+function isclade(tree::Phylogeny, vertex::Int)
+    return vertex > tree.ntaxa
+end
+
+
+
+"""
+    hasparent(tree::Phylogeny, vertex::Int)
+
+Check if a vertex in a phylogeny is linked to a parent.
+"""
+function hasparent(tree::Phylogeny, vertex::Int)
+    return indegree(tree.graph, vertex) == 1
+end
+
+
+"""
+    hasparent(tree::Phylogeny, name::SymString)
+
+Check if the vertex in a phylogeny is linked to a parent.
+"""
+function hasparent(tree::Phylogeny, name::SymString)
+    return hasparent(tree, tree.vertexindex[name])
+end
+
+
+"""
+    hasparent(tree::Phylogeny, vertices::AbstractVector{Int})
+
+Check some vertices in a phylogeny and test if each each is linked to a parent.
+"""
+function hasparent(tree::Phylogeny, vertices::AbstractVector{Int})
+    return indegree(tree.graph, vertices) .== 1
+end
+
+
+
+"""
+    noparent(tree::Phylogeny, vertex::Int)
+
+Check if a vertex in a phylogeny is not linked to a parent.
+"""
+function noparent(tree::Phylogeny, vertex::Int)
```

Comment:
That would also make it return true if the node had more than one parent - which should never happen, but coparent would return true instead of false if it ever did happen by mistake.